src/app/templates/InputGeneral/SelectGeneral.tsx

```diff
@@ -0,0 +1,219 @@
+  optionsProps,
+  testIds,
+  DropdownFaceContent,
+  optionsListClassName,
+  dropdownButtonClassName
+}: Props<T>) => {
+  const isInputDefined = isDefined(Input);
+  const { trackEvent } = useAnalytics();
+
+  const trackDropdownClick = () => {
+    if (testIds?.dropdownTestId) {
+      trackEvent(testIds?.dropdownTestId, AnalyticsEventCategory.DropdownOpened);
+    }
+  };
+
+  return (
+    <Popper
+      placement="bottom"
+      strategy="fixed"
+      modifiers={sameWidthModifiers}
+      fallbackPlacementsEnabled={false}
+      popup={({ opened, setOpened }) => (
+        <SelectOptions
+          optionsListClassName={optionsListClassName}
+          opened={opened}
+          setOpened={setOpened}
+          {...optionsProps}
+        />
+      )}
+    >
+      {({ ref, opened, toggleOpened }) => (
+        <div ref={ref as unknown as React.RefObject<HTMLDivElement>}>
+          {opened ? (
+            <SelectSearch className={dropdownButtonClassName} {...searchProps} />
+          ) : (
+            <div
+              className="w-full flex items-center justify-between border rounded-md border-gray-300"
+              style={{ maxHeight: '4.5rem', overflow: 'hidden' }}
+            >
+              <button
+                className={classNames(
+                  'flex gap-2 items-center',
+                  isInputDefined && 'border-r border-gray-300',
+                  !isInputDefined && 'w-full justify-between',
+                  dropdownButtonClassName
+                )}
+                onClick={() => {
+                  toggleOpened();
+                  trackDropdownClick();
+                }}
+                {...setTestID(testIds?.dropdownTestId)}
+              >
+                {DropdownFaceContent}
+                <ChevronDownIcon className="text-gray-600 stroke-current stroke-2" style={{ height: 16, width: 16 }} />
+              </button>
+              {Input}
+            </div>
+          )}
+        </div>
+      )}
+    </Popper>
+  );
+};
+interface SelectOptionsPropsBase<Type> {
+  options: Array<Type>;
+  noItemsText: ReactNode;
+  isLoading?: boolean;
+  optionsListClassName?: string;
+  onOptionChange: (newValue: Type) => void;
+  renderOptionContent: (option: Type) => ReactNode;
+}
+interface SelectOptionsProps<Type> extends SelectOptionsPropsBase<Type> {
+  opened: boolean;
+  setOpened: Dispatch<SetStateAction<boolean>>;
+}
+
+const SelectOptions = <Type extends unknown>({
+  opened,
+  options,
+  noItemsText,
+  isLoading,
+  optionsListClassName,
+  onOptionChange,
+  setOpened,
+  renderOptionContent
+}: SelectOptionsProps<Type>) => {
+  const handleOptionClick = (newValue: Type) => {
+    onOptionChange(newValue);
+    setOpened(false);
+  };
+
+  return (
+    <DropdownWrapper
+      opened={opened}
+      className="origin-top overflow-x-hidden overflow-y-auto"
+      style={{
+        maxHeight: '15.125rem',
+        backgroundColor: 'white',
+        borderColor: '#e2e8f0'
```

Comment:
move this to tailwind config please

Reply:
Select general has inner styles that we can't interrupt. This problem applies to the entire project. I propose to create a task in which we will fix it once and everywhere